src/masoniteorm/models/Model.py

```diff
@@ -461,11 +463,22 @@ def save(self, query=False):
 
         self.__dirty_attributes__.pop("builder")
 
+        self.observe_events(self, "saving")
+
         if not query:
-            return builder.update(self.__dirty_attributes__)
+            result = builder.update(self.__dirty_attributes__)
+            self.observe_events(self, "saved")
+            return result
 
         return builder.update(self.__dirty_attributes__, dry=True).to_sql()
 
+    def observe_events(self, model, event):
```

Comment:
What if remove the argument "model" from observers and uses "self": `self.__observers__` , `getattr(observer, event)`(self)

To be possible calling more cleaned:

```python
model.observe_events("event") 
```
Instead:

```python
model.observe_events(model,"event")
```

Reply:
great idea. Problem is that we need this same exact logic in 2 places. Some events happen only in the model (like hydrating) and some events happen in the query builder class (deleting, updating, etc). Because of this I abstracted the method out into a new class called `ObservesEvents` and inherited the model and query builder. To make it able to be used in both classes I need to accept a model

Reply:
Great refactoring.